01_fib/main.go

```diff
@@ -0,0 +1,25 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func main() {
+	fib(5)
+
+}
+
+func fib(n int) int {
+	var n1, n2, n3 = 0, 1, 0
+	for i := 0; i < n; i++ {
+		n1 = n2
+		n2 = n3
+		n3 = n1 + n2
```

Comment:
I think this should be the first assignment. Otherwise, `n2` becomes zero in the first pass. Also, you can do `n3 := n1 + n2` since n3 doesn't need to be tracked from one pass of the loop to the next. More so, you don't even need `n3` if you fold the sum into the one-liner suggested above.

Reply:
changed the logic to one-liner